### PR TITLE
build(deps): Upgrade CAPX version to v1.4.0

### PR DESCRIPTION
**Pull request by thunderboltsid**
Update version of CAPX to latest CAPX release v1.4.0

### CODE REVIEW

**Review by jimmidyson — APPROVED** [review]
Thank you!

File: pkg/handlers/nutanix/mutation/machinedetails/inject.go [hunk 1]
```diff
@@ -92,9 +92,9 @@ func (h *nutanixMachineDetailsPatchHandler) Mutate(
 
 			spec := obj.Spec.Template.Spec
 
-			spec.BootType = capxv1.NutanixBootType(nutanixMachineDetailsVar.BootType)
-			spec.Cluster = capxv1.NutanixResourceIdentifier(nutanixMachineDetailsVar.Cluster)
-			spec.Image = capxv1.NutanixResourceIdentifier(nutanixMachineDetailsVar.Image)
+			spec.BootType = nutanixMachineDetailsVar.BootType
```
Comment (dkoshkin) [inline pkg/handlers/nutanix/mutation/machinedetails/inject.go:95]: Thanks! I think I missed it as part of https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/698?

**Review by dkoshkin — APPROVED** [review]
👏

File: pkg/handlers/nutanix/mutation/machinedetails/inject.go (hunk 1 above)
Comment (jimmidyson) [reply pkg/handlers/nutanix/mutation/machinedetails/inject.go:95]: Yes, don't miss this again @dkoshkin 😛

**Pull request merged**